Automattic/newspack-ios
# Issues/75 folder contents data

Refs #75 

This PR is the first part of adding a core data proxy layer to story assets.  In this PR we:
- Roll back earlier temporary code that interacted directly with the file system
- Implement an AssetStore to wrangle the StoryAsset model
- Add the ability to create a rudimentary text note. This will be expanded on in its own focused sprint.
- Wire up basic display, create, and delete functionality for the AssetViewController

Future PRs will introduce sorting and ordering of the list, and core data/file system reconciliation. 

To test:
- View the list of story folders.
- Add some extra folders if necessary so you have more than one.
- Tap into a folder that is not the currently selected one (this should automatically select it).  You should now be viewing the Assets controller for the folder you selected.
- Use the plus button to add a few assets. Each asset should be shown as a row titled "New Text Note".
- Swipe to delete one of the rows. Confirm the delete works. 

@jleandroperez Can I trouble you with a review of this one?
